### ISSUE TITLE
Testing for complex WCAG criteria (target, focus appearance)

### ISSUE THREAD

**Issue by dav-idc** [alphagov/govuk-frontend#4347]

**Comment by dav-idc** [comment]
The testing sheet is now complete!
https://docs.google.com/spreadsheets/d/1PpKvhCGZJb1j0ZUUY2ZuA0CBQeu04yFD1bumiL8wGX8/edit?usp=sharing

Next step will be to glean potential Github issues that can be made from the 'focus appearance' entries (rows 31 to 61).

**Comment by dav-idc** [comment]
My final thoughts, for whomever picks this up:

- assign someone to take over, possibly beeps or an interaction designer
- definitely finish up those 3 tickets in the task list above ('WCAG 2.2 issues to resolve')
- decide what to do about the 30 ‘focus appearance’ entries near the bottom of that spreadsheet — there’s a ‘second 
- at a minimum I would strongly recommend we update our focus state or overall styling for:
    - secondary buttons
    - inverse buttons
    - tabs (the rectangular tab bit)
    - black links (back, breadcrumbs)

**Comment by owenatgov** [comment]
@querkmachine made [this comment](https://github.com/alphagov/govuk-frontend/issues/3839#issuecomment-1662131150) investigating all our focus states and how well they meet (or don't meet) 2.4.13: focus appearance.

Looking at the testing spreadsheet, it appears that David was slightly more lenient in his interpretation. Most notably for inline links (see the row in the calculator for the check answers pattern for an equivalent visual element to test) David has taken into account the contrast change of the text itself which means it just passes. For everything else, David's testing lines up with @querkmachine's testing.

@querkmachine has recommended that our focus states overall need a more thorough review. The issues I've added to the "issues to resolve" task list are potential quick wins but we should think about how we present our focus states more thoroughly via https://github.com/alphagov/govuk-design-system/issues/3007 at some point.
